### PR TITLE
Fixed bug: data structure did not preserve ordering.

### PR DESCRIPTION
**Pull request by LazyScholar**
#28

### CODE REVIEW

File: src/DocumenterCitations.jl [hunk 1]
```diff
@@ -11,11 +11,12 @@ using Documenter.Expanders
...
 using Bibliography
 using Bibliography: xnames, xyear, xlink, xtitle, xin
+using DataStructures: OrderedDict
 using Unicode
 
 export CitationBibliography
 struct CitationBibliography <: Documenter.Plugin
-    bib::Dict
+    bib::OrderedDict{String,Bibliography.Entry}
 end
```
Comment (ali-ramadhan) [inline src/DocumenterCitations.jl:20]: Out of curiousity would this be a bit more future proof?

```julia
struct CitationBibliography{B} <: Documenter.Plugin
    bib :: B
end
```

**Review by ali-ramadhan — APPROVED** [review]

File: src/DocumenterCitations.jl (hunk 1 above)
Comment (LazyScholar) [reply src/DocumenterCitations.jl:20]: I am not sure i know that some structs have the field types in their descriptions but i am too 'green' to see all implications/advantages.

Comment (LazyScholar) [reply src/DocumenterCitations.jl:20]: I tried it and it seems not to work.
It throws an error on https://github.com/ali-ramadhan/DocumenterCitations.jl/blob/8606536a10cd2c6328fd91cf0d4de180dad3b3a8/src/bibliography.jl#L42 with the error 
```
ERROR: LoadError: KeyError: key CitationBibliography not found
```

Comment (LazyScholar) [reply src/DocumenterCitations.jl:20]: I think it has only to be changed in the future if `Bibliography.jl` switches the data type to sorted dict, decides to change the entry structure to a abstract struct or if the keys are changed to not strings.

**Pull request merged**